simple_graph.py

```diff
@@ -8,65 +8,72 @@ def nodes(self):
         return self.graph.keys()
 
     def edges(self):
-        """Return a list of edges in the graph."""
+        """Return a list of edges and their weights in the graph."""
         l = []
         for k, v in self.graph.iteritems():
             for node in v:
-                l.append([k, node])
+                l.append([k, node, v[node]])
         return l
 
     def add_node(self, node):
         """Add a node to the graph."""
-        self.graph.setdefault(node, [])
+        self.graph.setdefault(node, {})
 
-    def add_edge(self, node1, node2):
+    def add_edge(self, node1, node2, weight):
         """Add an edge to the graph."""
         try:
-            self.graph.setdefault(node2, [])
-            self.graph[node1].append(node2)
+            self.graph.setdefault(node2, {})
+            self.graph[node1].setdefault(node2, weight)
         except KeyError:
-            self.graph.setdefault(node1, [node2])
+            self.graph.setdefault(node1, {node2: weight})
 
     def del_node(self, node):
-        """Delete a node from the graph. Throw a KeyError if the node wasn't
-        in the graph."""
+        """
+        Delete a node from the graph. Throw a KeyError if the node wasn't
+        in the graph.
+        """
```

Comment:
This should be a two-line docstring.  Starting at `Throw` there should be a new line